### PR TITLE
feat: default code offset = 3

### PR DESCRIPTION
**Pull request by bout3fiddy**
### What I did

Set default code offset for blueprint contracts to 3

### How I did it

Just change the default value from `zero_value` (which is `IRnode.from_list(0, typ=UINT256_T)`) to `IRnode.from_list(3, typ=UINT256_T)`.

### How to verify it

Added a test that simply uses no code_offset for initcode with the EIP5202 magic val appended at the top

### Commit message

Sets default code offset to 3 (instead of 0).

### Description for the changelog

Update: Sets default code offset to 3 (instead of 0).

### Cute Animal Picture

image

### CODE REVIEW

**Review by charles-cooper — COMMENTED** [review]
hmm i think this breaks some tests, also needs a docs update

**Comment by charles-cooper** [comment]
alright i fixed the tests and updated the docs. since this is a breaking change (and we don't really want to break user code with 'patch' changes), going to hold off on this until the 0.4.0 series.

also i'm not a huge fan of the duplicated code in the new test case - it would be nice to think of a way around that.

**Comment by charles-cooper** [comment]
i fixed the merge conflicts and updated the test so it passes. @pcaversaccio @fubuloubu want to take a look in case i missed anything?

File: tests/functional/builtins/codegen/test_create_functions.py [hunk 1]
```diff
@@ -146,9 +146,12 @@ def test(_salt: bytes32) -> address:
         c.test(salt, transact={})
 
 
+ERC5202_PREFIX = b"\xFE\x71\x00"
```
Comment (fubuloubu) [inline tests/functional/builtins/codegen/test_create_functions.py:149]: import from conftest?

**Review by fubuloubu — APPROVED** [review]

File: tests/functional/builtins/codegen/test_create_functions.py (hunk 1 above)
Comment (charles-cooper) [reply tests/functional/builtins/codegen/test_create_functions.py:149]: it should be in vyper/utils.py really

Comment (charles-cooper) [reply tests/functional/builtins/codegen/test_create_functions.py]: https://github.com/vyperlang/vyper/pull/3454/commits/afa1af04e7a98928914f9394d89244fe1b85c0f5

File: docs/built-in-functions.rst
Comment (pcaversaccio) [inline docs/built-in-functions.rst:1]: Can we add a note here about this breaking change. E.g.

```rst
.. note::

  Until Vyper version ``0.3.10``, the `code_offset` defaulted to ``0``.
```

File: tests/functional/builtins/codegen/test_create_functions.py [hunk 2]
```diff
@@ -238,8 +298,8 @@ def test(code_ofst: uint256) -> address:
     tx_info = {"from": w3.eth.accounts[0], "value": 0, "gasPrice": 0}
     tx_hash = deploy_transaction.transact(tx_info)
     blueprint_address = w3.eth.get_transaction_receipt(tx_hash)["contractAddress"]
-    blueprint_code = w3.eth.get_code(blueprint_address)
-    print("BLUEPRINT CODE:", blueprint_code)
+    # blueprint_code = w3.eth.get_code(blueprint_address)
```
Comment (pcaversaccio) [inline tests/functional/builtins/codegen/test_create_functions.py:301]: We can safely delete these 2 commented lines IMO

**Review by pcaversaccio — APPROVED** [review]

**Pull request merged**